### PR TITLE
[PATCH v3] doc: userguide: update code blocks

### PR DESCRIPTION
**Pull request by MatiasElo**
Remove code sections copied from the API headers as they will not stay up- to-date. Update some code examples to utilize current APIs and remove some obsolete ones.

### CODE REVIEW

File: doc/users-guide/users-guide-pktio.adoc [hunk 1]
```diff
@@ -136,330 +61,26 @@ struct, in turn, specifies the input and output *modes* of the PktIO.
...
-	  *
-	  * The default value is ODP_PROTO_LAYER_ALL. */
-	odp_proto_layer_t layer;
-
-} odp_pktio_parser_config_t;
------
-Note that parsing is automatically done whenever classification is enabled
-for an RX interface (see below).
+receipt as controlled by the `odp_pktio_parser_config_t` struct. Note that
+parsing is automatically done whenever classification is enabled for an RX
```
Comment (JannePeltonen) [inline doc/users-guide/users-guide-pktio.adoc:76]: > parsing is automatically done whenever classification is enabled

This statement caught my eye even though it is not changed by this commit. I wonder if enabling of parsing is really guaranteed by the API in this case as I did not find it in the API text. Or perhaps this is just the way the linux-gen ODP implementation works?

File: doc/users-guide/users-guide-pktio.adoc [hunk 2]
```diff
@@ -500,97 +121,7 @@ of type *odp_pktin_queue_t* and are retrieved by threads calling the
...
-	  * odp_queue_param_t documentation.
-	  * When classifier is enabled in odp_pktin_queue_config() this
-	  * value is ignored. */
-	odp_queue_param_t queue_param;
-
-} odp_pktin_queue_param_t;
------
-Note that the *queue_param* field of this struct is ignored in DIRECT mode.
-The purpose of `odp_pktin_queue_config()` is to specify the number of PktIn
+`odp_pktin_queue_config()` API, which purpose is to specify the number of PktIn
```
Comment (JannePeltonen) [inline doc/users-guide/users-guide-pktio.adoc:124]: The 'which' does not sound quite right to me. I would probably use 'whose', or if that is not good, I would say "... API, the purpose of which is ...".

**Review by JannePeltonen — APPROVED** [review]
I made two comments, one not really related to the changes donen here and the other so minor that feel free to ignore it.

File: doc/users-guide/users-guide-pktio.adoc (hunk 1 above)
Comment (MatiasElo) [reply doc/users-guide/users-guide-pktio.adoc:76]: Indeed the API doesn't mention this, so removed the sentence.

**Pull request merged**